lib/pf/person.pm

```diff
@@ -375,6 +388,42 @@ sub person_violations {
     return db_data(PERSON, $person_statements, 'person_violations_sql', $pid);
 }
 
+=head2 persons_without_nodes
+
+Get all the persons who are not the owner of at least one node.
+
+=cut
+
+sub persons_without_nodes {
+   return db_data(PERSON, $person_statements, 'persons_without_nodes_sql'); 
+}
+
+=head2 person_cleanup
+
+Clean all persons that are not the owner of a node and that are not a local account that is still valid
+
+=cut
+
+sub person_cleanup {
+    my @to_delete = map { $_->{pid} } persons_without_nodes();
+    my $now = DateTime->now();
+    foreach my $pid (@to_delete) {
+        my $password = pf::password::view($pid);
+        if(defined($password)){
+            my $expiration = DateTime::Format::MySQL->parse_datetime($password->{expiration});
+            my $cmp = DateTime->compare($now, $expiration);
+            if($cmp < 0){
+                get_logger->debug("Not deleting $pid because the local account is still valid.");
+                next;
+            }
+            # We delete the password too
+            pf::password::_delete($pid);
+        }
+        # We're all good for deletion
+        person_delete($pid);
+    }
+}
+
```

Comment:
There is none either in the node cleanup and it works perfectly fine.

The node and person tables are less than a few tens of thousands of rows.

My point is there is no need to engineer a solution for this specific case.

We should instead address the problem in general using the database-cleaner logic and use it in all of our cleanups, not just do it for the case where it is almost useless